### PR TITLE
Gate refactoring

### PR DESCRIPTION
**Pull request by jokertyf**

### CODE REVIEW

File: src/levels/Level.gd [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class_name Level
+extends Node2D
+
```
Comment (thearst3rd) [inline src/levels/Level.gd:3]: reeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee

File: src/objects/Gate.gd [hunk 2]
```diff
@@ -0,0 +1,32 @@
+extends Node2D
+
+
+var tile_map: TileMap
+var grid_coords: Vector2
+
```
Comment (thearst3rd) [inline src/objects/Gate.gd:6]: reeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee

File: src/objects/Gate.gd [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		z_index = 1
+
+
+func open() -> void:
+	$AnimatedSprite.play("open")
+	$AnimatedSprite.offset.y += 8
+	tile_map.set_cellv(grid_coords, tile_map.tile_set.find_tile_by_name("Base"))
+
+
+func	 close() -> void:
```
Comment (thearst3rd) [inline src/objects/Gate.gd:29]: is there a tab in there somehow lol

**Review by thearst3rd — CHANGES_REQUESTED** [review]
pedantry

**Review by thearst3rd — APPROVED** [review]

**Review by kdreese — APPROVED** [review]

**Pull request merged**